### PR TITLE
fix(relay/client): only try to forward handler events once

### PR DESCRIPTION
**Pull request by elenaf9**
## Description

We currently try to forward pending handler events in the relay client behavior twice: in `handle_established_{in, out}bound_connection` and in `on_swarm_event()` (introduced in #3328).
This is redundant because by the second time we try to forward it there won't be any pending event anymore.
This PR removes the duplicated logic from `on_swarm_event`. Forwarding it in `handle_established_{in, out}bound_connection` is more convenient as we still have direct access to the handler there and don't need to go through the swarm.

However, it requires that in `handle_established_{in, out}bound_connection` the event is also removed when the connection is relayed, in which case the event is simply discarded because the commands are only valid on direct connections.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [ ] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: protocols/relay/src/priv_client.rs [hunk 1]
```diff
@@ -168,12 +168,14 @@ impl NetworkBehaviour for Behaviour {
...
     ) -> Result<THandler<Self>, ConnectionDenied> {
+        let pending_handler_command = self.pending_handler_commands.remove(&connection_id);
+
         if local_addr.is_relayed() {
             return Ok(Either::Right(dummy::ConnectionHandler));
         }
         let mut handler = Handler::new(self.local_peer_id, peer, remote_addr.clone());
 
-        if let Some(event) = self.pending_handler_commands.remove(&connection_id) {
+        if let Some(event) = pending_handler_command {
```
Comment (jxs) [inline protocols/relay/src/priv_client.rs:178]: nit: I feel it was more readable the previous way as it was all on a single line. You mention that:
>However, it requires that in handle_established_{in, out}bound_connection the event is also removed when the connection is relayed, in which case the event is simply discarded because the commands are only valid on direct connections.

but this happened before right?

**Review by jxs — APPROVED** [review]
Nice catch Elena LGTM!

File: protocols/relay/src/priv_client.rs (hunk 1 above)
Comment (elenaf9) [reply protocols/relay/src/priv_client.rs:178]: > but this happened before right?

I am not sure what you mean. Could you rephrase? :)

We now need to remove the pending handler commands before the early return when `local_addr.is_relayed()`. That's why I moved it to the top of the function, and have to use the resulting `pending_handler_command` here.

Comment (jxs) [reply protocols/relay/src/priv_client.rs:178]: makes sense, thanks for explaining Elena

**Review by jxs — APPROVED** [review]
